airbyte-integrations/connectors/source-marketo/source_marketo/source.py

```diff
@@ -300,8 +303,12 @@ def path(self, **kwargs) -> str:
     def should_retry(self, response: requests.Response) -> bool:
         if response.status_code == 429 or 500 <= response.status_code < 600:
             return True
-        record = next(self.parse_response(response, {}), {})
-        status, export_id = record.get("status", "").lower(), record.get("exportId")
+        if errors := response.json().get("errors"):
+            if errors[0].get("code") == "1029" and re.match("Export daily quota \d+MB exceeded", errors[0].get("message")):
+                message = "Daily limit for job extractions has been reached (resets daily at 12:00AM CST)."
+                raise AirbyteTracedException(internal_message=response.text, message=message, failure_type=FailureType.config_error)
+        result = response.json().get("result")[0]
```

Comment:
nit: this construction looks a little bit strange for me. You are getting `result` using safe `.get` method which means you expect respone may not have `result` in response body. But at the same time you are getting first element `[0]` of it like you expect result is allways present. I think it will be more clear if we replace it with this: 
```
response.json()["result"][0]
```
if you are sure `result` will allways be present in response, or this:
```
response.json().get("result", [None])[0]
```